app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -348,17 +355,103 @@ private void setRoots() {
     }
 
     private int initializeAppResources() {
-        //There should be exactly one of these for now
+        // Before we try to initialize a new app, check if any existing apps were left in a
+        // partially deleted state, and finish uninstalling them if so
         for (ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
-            if (record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
-                //We have an app record ready to go
+            if (record.getStatus() == ApplicationRecord.STATUS_DELETE_REQUESTED) {
+                record.uninstall(this);
+            }
+        }
+        
+        // There may now be multiple app records in storage, because of multiple apps support.
+        // Since this method is called from onCreate of CommCareApplication, current behavior is
+        // just to pick the first app in the list to initialize
+        for(ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
+            if(record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
                 return initializeAppResources(new CommCareApp(record));

```

Comment:
Hey,

The "Current" app is stored in the preferences, right? It seems like we could poll through to find that one before we do the full initialization here, I think. 

This is an area that will need to be revisited int he future anyway, because this initialization is what makes the app modally non-responsive-with-a-white-background each time you cold start it, but int he short term we could just store the "favorite" app in the global shared preferences, which shhould work, or select the first if nothing is stored.